src/slam_karto.cpp

```diff
@@ -291,12 +317,17 @@ SlamKarto::~SlamKarto()
     transform_thread_->join();
     delete transform_thread_;
   }
+  if(vis_thread_)
+  {
+    vis_thread_->join();
+    delete vis_thread_;
+  }
   if (scan_filter_)
     delete scan_filter_;
   if (scan_filter_sub_)
     delete scan_filter_sub_;
   if (solver_)
-    delete solver_;
+    delete solver_.get();
```

Comment:
Good point, that was actually a copy/paste of our current indigo-devel branch, I didn't realize it was a smart pointer. That looks like a good candidate for the cause of https://github.com/savioke/savi_nav/issues/971